### ISSUE TITLE
[4.x] Error changing Date Fieldtype values

### ISSUE THREAD

**Issue by jackmcdade** [statamic/cms#8026]
### Bug description

I can't change date fields, and I get a bunch of Vue errors while trying.

<img width="872" alt="CleanShot 2023-04-29 at 09 56 44@2x" src="https://user-images.githubusercontent.com/44739/235306653-ac52e291-f692-49bc-ae86-2eb3b5c0d156.png">

### How to reproduce

Change a date field while on `master`

### Environment

```yaml
v4 beta
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

None

**Comment by o1y** [comment]
The mentioned error can be fixed by changing the following line in[ SinglePopover.vue](https://github.com/statamic/cms/blob/master/resources/js/components/fieldtypes/date/SinglePopover.vue#L86):

```diff
popoverStateChanged(open) {
      this.open = open;
-    this.$nextTick(() => this.resetPicker());
+    this.resetPicker()
},
```

But it seems there are more issues related to the Date Fieldtype (both single and range):
* Changing the date using the input field seems not updating the calendar value
* Clearing the input field is not clearing the date
* The ESC key is not hiding the date picker

**Comment by jasonvarga** [comment]
Fixed by #8036
